history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
+
+func NewHistory(m *MerkleTree, key crypto.SigningKey, startEp, epInterval int64) *History {
+	h := new(History)
+	h.tree = m
+	h.epochInterval = epInterval
+	h.currentSTR = newSTR(m, startEp, 0, make([]byte, crypto.HashSizeByte), key)
+	return h
+}
+
+func (h *History) UpdateHistory(m *MerkleTree, nextEp int64) error {
```

Comment:
> I may be reading this wrong, but as I understand it, an epoch is discrete. The current epoch is really the most recent snapshot that's been captured. Remember, in between epochs (including the first) we hand out temporary bindings, which aren't yet included in the str.

Yes. That's right. But how's about my question: 
> we have 2 different STRs of each epoch, right?

I'm asking it because I have some concerns related to the monitoring protocol.